### ISSUE TITLE
Non standard serialisation of integer keys (Convertion to String type needed before serialization)

### ISSUE THREAD

**Issue by MrDaiki** [simd-lite/simd-json#304]
Currently serializing a HashMap with integer (i32) keys using serde Serialize option produce non standard json outputs. In json, it is not possible to have integer as keys for object, but only string (see [https://www.json.org/json-en.html](https://www.json.org/json-en.html))

My code look something like this : 
```rust
#[derive(Serialize,Deserialize)]
pub struct ArticleData {
    data : HashMap<i32,usize>,
    ...
}
```
To serialize, the used line of code is : 
```rust
let file = File::create(filepath).unwrap();
let buf = std::io::BufWriter::new(file);
simd_json::to_writer_pretty(buf, &self).expect("Error");
```

and produce the following json (I only show a part of the file since I am working on heavy file workload) : 

```json
{
    "data": {
        124563008 : 2,
        470013323 : 1,
    ....

```

When correct standard output should be : 
```json
{
    "data": {
        "124563008" : 2,
        "470013323" : 1,
    ....

```

A possible workaround is to convert integer to string before serialization like that : 
```rust
#[derive(Serialize,Deserialize)]
pub struct ArticleData {
    data : HashMap<String,usize>,
    ...
}
```

**Comment by Licenser** [comment]
Looking into this, it is limited to the "pretty printing" serializer so a quick workaround would be using the normal one, but this'll be fixed today so probably no workaround needed :)

**Comment by Licenser** [comment]
released as `v0.10.5`
